kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldContainAllTest.kt

```diff
@@ -122,15 +122,15 @@ class ShouldContainAllTest : WordSpec() {
 
          "print one possible match for one mismatched element" {
             shouldThrowAny {
-               listOf(sweetGreenApple, sweetGreenPear, sourYellowLemon).shouldContainAll(
-                  listOf(sweetGreenApple, sweetRedApple)
+               listOf(sweetGreenPear, sweetGreenApple, sourYellowLemon).shouldContainAll(
+                  listOf(sweetGreenPear, sweetRedApple)
```

Comment:
Unfortunately, here both the assertion and the expected output have both changed, which ,makes it more difficult to review. Can you revert just the assertion back to 
```kotlin
 shouldThrowAny {
               listOf(sweetGreenApple, sweetGreenPear, sourYellowLemon).shouldContainAll(
                  listOf(sweetGreenApple, sweetRedApple)
```
so that the only this that changes is the expected message?

Reply:
Ok, I have now adjusted the test so that it runs successfully and only output has been adjusted. Does the name “print one possible match for one mismatched element” still make sense now?

Reply:
There are actually two differences. That was also the reason why I adjusted the assertion to have only one difference to the exception. This in turn caused the problem I described above!

Reply:
> There are actually two differences. That was also the reason why I adjusted the assertion to have only one difference to the exception. This in turn caused the problem I described above!

Now I understand why you made that change to the assertion itself, and I do agree that your change is an improvement. Yes, let's have the assertion as you originally suggested, because it's a much cleaner test:
```
istOf(sweetGreenPear, sweetGreenApple, sourYellowLemon).shouldContainAll(
                  listOf(sweetGreenPear, sweetRedApple)
```

So sorry about this misunderstanding.

Reply:
Thank you for your quick answers!

> So sorry about this misunderstanding.

No problem at all.

But unfortunately it's not enough to rewrite the test. This is the second part I have described. 
If one use

```kotlin
istOf(sweetGreenPear, sweetGreenApple, sourYellowLemon).shouldContainAll(
                  listOf(sweetGreenPear, sweetRedApple)
```

then the result is:

```
Actual was:
"Collection should contain all of [Fruit(name=pear, color=green, taste=sweet), Fruit(name=apple, color=red, taste=sweet)] but was missing [Fruit(name=apple, color=red, taste=sweet)]Possible matches:
 expected: Fruit(name=apple, color=red, taste=sweet),
  but was: Fruit(name=pear, color=green, taste=sweet),
  The following fields did not match:
    "name" expected: <"apple">, but was: <"pear">
        "color" expected: <"red">, but was: <"green">

 expected: Fruit(name=apple, color=red, taste=sweet),
  but was: Fruit(name=apple, color=green, taste=sweet),
  The following fields did not match:
    "color" expected: <"red">, but was: <"green">"
```

And that seems to be wrong or not? The first one (`but was: Fruit(name=pear, color=green, taste=sweet)`) shouldn't show up, should it?

And if it should, why is `sourYellowLemon` not displayed?

Reply:
> Thank you for your quick answers!
> 
> > So sorry about this misunderstanding.
> 
> No problem at all.
> 
> But unfortunately it's not enough to rewrite the test. This is the second part I have described. If one use
> 
> ```kotlin
> listOf(sweetGreenPear, sweetGreenApple, sourYellowLemon).shouldContainAll(
>                   listOf(sweetGreenPear, sweetRedApple)
> ```
> 
> then the result is:
> 
> ```
> Actual was:
> "Collection should contain all of [Fruit(name=pear, color=green, taste=sweet), Fruit(name=apple, color=red, taste=sweet)] but was missing [Fruit(name=apple, color=red, taste=sweet)]Possible matches:
>  expected: Fruit(name=apple, color=red, taste=sweet),
>   but was: Fruit(name=pear, color=green, taste=sweet),
>   The following fields did not match:
>     "name" expected: <"apple">, but was: <"pear">
>         "color" expected: <"red">, but was: <"green">
> 
>  expected: Fruit(name=apple, color=red, taste=sweet),
>   but was: Fruit(name=apple, color=green, taste=sweet),
>   The following fields did not match:
>     "color" expected: <"red">, but was: <"green">"
> ```
> 
> And that seems to be wrong or not? The first one (`but was: Fruit(name=pear, color=green, taste=sweet)`) shouldn't show up, should it?
> 
> And if it should, why is `sourYellowLemon` not displayed?

I concur, `sweetGreenPear` should not show up, as it matches only on one field, so it's just noise - this is an issue we should fix. But this issue is unrelated to your change, which (your change) makes sense and should be merged.

I'll take care of the issue with `sweetGreenPear` in the output, so that the output will be only

```kotlin
"Collection should contain all of [Fruit(name=pear, color=green, taste=sweet), Fruit(name=apple, color=red, taste=sweet)] but was missing [Fruit(name=apple, color=red, taste=sweet)]Possible matches:
 expected: Fruit(name=apple, color=red, taste=sweet),
  but was: Fruit(name=apple, color=green, taste=sweet),
  The following fields did not match:
    "color" expected: <"red">, but was: <"green">"
```

but that will be in a separate PR , later. let's merge this PR as is, and fix this test later.